src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -155,14 +158,41 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         return true;
     }
 
+    public static Map<String, TestClass>  findTestResultsInDirectory(Run<?,?> build, TaskListener listener, FilePath workspace){
+        if(workspace==null){
+            return Collections.emptyMap();
+        }
+        String[] tests = null;
+        Map<String, TestClass> data = new TreeMap<String, TestClass>();
+        final String baseDir = workspace.getRemote();
+        try {
+            tests = workspace.act(new MasterToSlaveCallable<String[], Throwable>() {
+                @Override
+                public String[] call() throws Throwable {
+
+                    return Util.createFileSet(new File(baseDir), "**/src/test/java/**/*Test.java").getDirectoryScanner().getIncludedFiles();
```

Comment:
this does not match the defaults that surefire uses so many tests will be missed.

http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#includes lists

```
<includes>
    <include>**/Test*.java</include>
    <include>**/*Test.java</include>
    <include>**/*Tests.java</include>
    <include>**/*TestCase.java</include>
</includes>
```

Reply:
I though about it, but I did not want to add there something what should not be there (what is not test), for example this one [1]. My first though was to have at least something in the first run (even if it means not to be effective and run some test in every split) and consult the more accurate version in different pullrequest. But if you agree to have it even if it means that we can catch something more then only tests (it is not like the maven cares about it in the end or it can affect a result or performance), I totally agree with this idea.

[1] https://github.com/jenkinsci/junit-plugin/blob/master/src/test/java/hudson/tasks/test/helper/TestResultsPage.java

Reply:
well if you are using exclusions you would run the same tests multiple times for the tests you do not find.
for inclusions you would fail to run some tests.

both cases are bad and if you found too much it should be much less of an issue than not finding enough, unless you happened to have a bucket with some files but actually contained no tests, which would cause a build failure.